db/rdbms-schema/src/main/resources/db/changelog/rdbms-exporter/changesets/8.8.0.xml

```diff
@@ -539,4 +539,26 @@
     </modifySql>
   </changeSet>
 
+  <changeSet id="create_sequence_flow_table" author="danielkelemen">
+    <createTable tableName="${prefix}SEQUENCE_FLOW">
+      <!-- no PK possible, flowNodeId can occur multiple times with multiInstanceLoopCharacteristics -->
+      <column name="FLOW_NODE_ID" type="VARCHAR(255)"/>
+      <column name="PROCESS_INSTANCE_KEY" type="BIGINT"/>
+      <column name="PROCESS_DEFINITION_KEY" type="BIGINT"/>
+      <column name="PROCESS_DEFINITION_ID" type="VARCHAR(255)"/>
+      <column name="TENANT_ID" type="VARCHAR(255)"/>
+      <column name="PARTITION_ID" type="NUMBER"/>
+      <column name="HISTORY_CLEANUP_DATE" type="TIMESTAMP WITH TIME ZONE(3)"/>
+    </createTable>
```

Comment:
💭 ES/OS uses piKey + fnid as key but it can't be a primary key because with multi instance characteristics the sequence flow is taken multiple times. The result needs to match ES/OS so RDBMS needs to return these sequence flows once only.

We could either:
1. Still use it as a PK but somehow prevent multiple inserts
   - Cons: No generic SQL upsert statement to achieve this.
2. Not use a PK, insert all sequence flows and return distinct results for the API.
   - Cons: Table without PK.

I decided to use option 2 due to simplicity.

Reply:
Interesting. For multi-instance scenarios, this will write quite a bit of data into this table that we'll never need, right?

Reply:
Yes, that's correct.
We only return 1 sequenceFlow in API for these.

Reply:
I can double check if there are supported options for insert ignore in different DBs, then we could use that with PK instead. IIRC, not all supported it.

Reply:
It's just a thought, but maybe we could programmatically add an UPSERT option from our application code that does a `IF NOT EXISTS(QUERY) Then INSERT`? If I recall correctly, the INSERT IGNORE has some issues as it swallows ANY issues that can occur upon INSERT. There could be many other issues where we'd want the exception back. This would be a key-based lookup, so it should be good enough, maybe?

But there could be some performance issues with this approach since we'll do this very often, I believe. Simply writing more data to the database could be less impactful, in the end, since only multi-instance uses cases do this.

Reply:
That's also my concern, it would be better to have a 1 step DB level upsert instead of application level select then insert.

Reply:
Something like this could help but it's highly database specific, so 🤷
https://northconcepts.com/blog/2023/09/19/upsert-relational-databases/

Reply:
Nice!